### PR TITLE
Changestreams db and cluster level #251

### PR DESCRIPTION
**Pull request by peterdeka**
Enhancing change streams in order to support DB level and cluster level changes notifications introduced in mongoDB 4.0 as per issue #251.
I strongly ask review of the way i addressed the question as i'm not really sure it is not a workaround.
If we go this way, i think it's better to make unexported all session and database methods that were added (`Pipe()`, `NewIter()`), except for the `Watch()` method in order to not pollute with meaningless methods.
**Tests are still missing** and will be added once we agree on the way i implemented the functionality and Mongo 4.0 is added to the test suite.

### CODE REVIEW

**Comment by peterdeka** [comment]
any news on this @eminano i just need a fast verification if my solution is viable in order to consolidate the PR completing the missing points.

Thanks, 
Pietro

**Comment by eminano** [comment]
Hi @peterdeka,

As @domodwyer mentioned on issue #251 this looks like a reasonable approach for the implementation. Will wait for you to complete the PR before doing a full review.

Thanks!
Esther

**Comment by peterdeka** [comment]
For me this is ok, i'm going on with the tests. But then we will need 4.0 to be in the suite to make sense of them.

**Comment by domodwyer** [comment]
Hi @peterdeka 

If you have a branch with working cluster-level changestreams, just change the [`.travis.yml`](https://github.com/globalsign/mgo/blob/master/.travis.yml) file in your PR to include 4.0 and it'll automatically run!

Dom

**Comment by peterdeka** [comment]
@eminano i'm good to go, Travis tests are obviously excluding 4.0 but i ran locally and they are green.

**Comment by eminano** [comment]
@peterdeka, could you please resolve the conflicts? We'll review once the build is ready.

Thanks! 
Esther

**Comment by peterdeka** [comment]
Just for the history books, currently blocked by #266

File: changestreams.go [hunk 1]
```diff
@@ -17,6 +17,14 @@ const (
 	UpdateLookup = "updateLookup"
 )
 
+type ChangeDomainType int
```
Comment (domodwyer) [inline changestreams.go]: As this is an implementation detail (never exposed / used by an `mgo` user) we should probably not export it, or it's variants below.

**Comment by domodwyer** [comment]
@peterdeka this is a really, really good PR - thanks so much for the contribution!

I left a single comment regarding the visibility of `ChangeDomainType` - I see no reason this can't be merged functionally!

Thanks very much -  I know the mgo code base can be difficult to work with but this is a great PR.

Dom

File: changestreams.go (hunk 1 above)
Comment (peterdeka) [reply changestreams.go]: Ok that makes sense, i'll do it ASAP.

Thanks @domodwyer

Comment (peterdeka) [reply changestreams.go]: Fixed it @domodwyer .
As usual @eminano, travis randomly failed a build... :D

**Comment by eminano** [comment]
Thanks @peterdeka for a great PR!

**Pull request merged**